src/osd/OSD.cc

```diff
@@ -5599,8 +5600,12 @@ void OSD::send_beacon(const ceph::coarse_mono_clock::time_point& now)
         ceph::features::mon::FEATURE_LUMINOUS)) {
     dout(20) << __func__ << " sending" << dendl;
     last_sent_beacon = now;
-    auto beacon = new MOSDBeacon(osdmap->get_epoch(), min_last_epoch_clean);
-    beacon->pgs = std::move(min_last_epoch_clean_pgs);
+    MOSDBeacon* beacon = nullptr;
+    {
+      Mutex::Locker l{min_last_epoch_clean_lock};
+      beacon = new MOSDBeacon(osdmap->get_epoch(), min_last_epoch_clean);
+      beacon->pgs = std::move(min_last_epoch_clean_pgs);
```

Comment:
swap() instead?  i don't really understand what the post-condition of std::move is.  here we want min_last_epoch_clean_pgs to be valid but empty, but in other cases the object is unusable